src/Microsoft.DotNet.Darc/src/DarcLib.AzDev/AzureDevOpsClient.cs

```diff
@@ -446,6 +447,75 @@ public async Task MergePullRequestAsync(string pullRequestUrl, MergePullRequestP
+        {
+            (string accountName, string projectName, string repoName, int id) = ParsePullRequestUri(pullRequestUrl);
+            VssConnection connection = CreateVssConnection(accountName);
+            GitHttpClient client = await connection.GetClientAsync<GitHttpClient>();
+            var prInfo = await client.GetPullRequestAsync(repoName, id, includeCommits: true);
+            IList<Commit> commits = new List<Commit>(prInfo.Commits.Length);
+            for (int i=0; i < prInfo.Commits.Length; i++)
+            {
+                commits.Add(new Commit(prInfo.Commits[i].Author.Name == "DotNet-Bot"? "dotnet-maestro[bot]" : prInfo.Commits[i].Author.Name, 
+                    prInfo.Commits[i].CommitId ,
+                    prInfo.Commits[i].Comment));
+            }
+            return commits;
+        }
+
+        /// <summary>
+        ///     Merges an update dependency pull request
+        /// </summary>
+        /// <param name="pullRequestUrl">Uri of the update dependency pull request to merge</param>
+        /// <param name="parameters">Settings for merge</param>
+        /// <returns></returns>
+        public async Task MergeDependencyPullRequestAsync(string pullRequestUrl, MergePullRequestParameters parameters, string commitToMerge)
+        {
+            (string accountName, string projectName, string repoName, int id) = ParsePullRequestUri(pullRequestUrl);
+
+            VssConnection connection = CreateVssConnection(accountName);
+            GitHttpClient client = await connection.GetClientAsync<GitHttpClient>();
+            var prInfo = await client.GetPullRequestAsync(repoName, id , includeCommits:true);
+
+            await client.UpdatePullRequestAsync(
+                new GitPullRequest
+                {
+                    Status = PullRequestStatus.Completed,
+                    CompletionOptions = new GitPullRequestCompletionOptions
+                    {
+                        MergeCommitMessage = commitToMerge,
+                        BypassPolicy = true,
+                        BypassReason = "All required checks were successful",
+                        SquashMerge = parameters.SquashMerge,
+                        DeleteSourceBranch = parameters.DeleteSourceBranch
+                    },
+                    LastMergeSourceCommit = new GitCommitRef { CommitId = prInfo.LastMergeSourceCommit.CommitId, Comment = commitToMerge }
+                },
+                projectName,
+                repoName,
+                id);
+        }
+
+        /// <summary>
+        ///     Parse out the owner and repo from a repository url
+        /// </summary>
+        /// <param name="prBody">Pull Request commit msg</param>
+        /// <returns>Tuple of owner and repo</returns>
+        public static string ParsePullRequestBody(Regex pattern, string prBody)
+        {
+            var matches = pattern.Matches(prBody);
```

Comment:
This is a very strange use of regular expressions.  It looks like it's just trying to extract whatever matches the pattern, but without the "*" and gluing them all together with newlines?  That pretty easy with Regex itself, we don't need  all these replaces.

The regex's used here have a bunch of names groups, but those groups are never used... They could be massively simplified to capture the stuff we want, then this whole method would just be `pattern.Matches.Select(m => m.Group[1].Value)...`  At which point, it doesn't really need to be a method, and it's less weird to try and figure out what regex's this expects and how it will behave.

Reply:
I'm... not 100% sure what this method is doing. It's name very, very generically, and definitely doesn't do what the comment says it does.  It doesn't return the owner at all.  It just returns the regular expression, but with the bold bits removed.  Why is it removing the bold bits?  It doesn't sound like it should be returning a string at all, but rather a struct that's Owner/Repository.

Reply:
Redid this using a different regex now.